sentry-rails/lib/sentry/rails/active_job.rb

```diff
@@ -57,13 +57,36 @@ def already_supported_by_sentry_integration?
       def sentry_context
         {
           active_job: self.class.name,
-          arguments: arguments,
+          arguments: sentry_serialize_arguments(arguments),
           scheduled_at: scheduled_at,
           job_id: job_id,
           provider_job_id: provider_job_id,
           locale: locale
         }
       end
+
+      def sentry_serialize_arguments(arguments)
+        arguments.map do |argument|
+          sentry_serialize_argument(argument)
+        end
+      end
+
+      def sentry_serialize_argument(argument)
+        case argument
+        when Hash
+          argument.reduce({}) do |h, (k, v)|
+            h.merge(k => sentry_serialize_argument(v))
+          end
+        when Array, Enumerable
+          argument.map { |v| sentry_serialize_argument(v) }
+        when ->(v) { v.respond_to?(:to_global_id) }
+          argument.to_global_id.to_s
+        else
+          argument
+        end
+      rescue StandardError
+        argument
+      end
```

Comment:
Presuming we have Active Support available here in sentry-rails, we could perhaps simplify the implementation with [#deep_transform_values](https://edgeapi.rubyonrails.org/classes/Hash.html#method-i-deep_transform_values)

Reply:
Godo point. I think this code is roughly based on my initial issue wherein I didn't want to assume ActiveSuport, but given this is would be in the `-rails` gem, seems fine to assume it's there.

Reply:
Well I did simplified the implementation to:

```rb
      def sentry_serialize_arguments(argument)
        case argument
        when Hash
          argument.deep_transform_values { |v| sentry_serialize_arguments(v) }
        when Array, Enumerable
          argument.map { |v| sentry_serialize_arguments(v) }
        when ->(v) { v.respond_to?(:to_global_id) }
          argument.to_global_id.to_s
        else
          argument
        end
      rescue StandardError
        argument
      end
```

But it turns out `deep_transform_values` is added after Rails 6.0. And we're still supporting Rails 5.*, so....

Reply:
Ah ok 😅

Reply:
@lewispb fyi `transform_values` also works 😄

Reply:
@lewispb fyi `transform_values` also works 😄